examples/provekit-shim-python-sqlite3/provekit_shim_python_sqlite3/__init__.py

```
@@ -617,6 +617,60 @@ def connection_row_factory_callable(conn: sqlite3.Connection, factory: Callable)
+# The free `db`/`cursor` names pass through the param->placeholder projection
+# unchanged; only `sql`/`args` map to ${param0}/${param1}.
+
+@sugar.bind(
+    concept="concept:sql-execute",
+    library="sqlite3",
+    family="concept:family:sql",
+    version="python-3",
+    loss=["sync-vs-async", "last-insert-id", "transaction-isolation", "row-typing-mode"],
+)
+def migrate_execute(sql, args):
+    cursor = db.execute(sql, tuple(args))
```

Ruff (0.15.13)

[error] Undefined name `db`

(F821)